ui/imports/shared/popups/ImageCropWorkflow.qml

```diff
@@ -39,14 +42,37 @@ Item {
 
         title: root.imageFileDialogTitle
         folder: root.userSelectedImage ? imageCropper.source.substr(0, imageCropper.source.lastIndexOf("/")) : shortcuts.pictures
-        nameFilters: [qsTr("Supported image formats (%1)").arg(Constants.acceptedDragNDropImageExtensions.map(img => "*" + img).join(" "))]
+        nameFilters: [qsTr("Supported image formats (%1)").arg(UrlUtils.validImageNameFilters)]
         onAccepted: {
             if (fileDialog.fileUrls.length > 0) {
-                cropImage(fileDialog.fileUrls[0])
+                const url = fileDialog.fileUrls[0]
+                if (Utils.isValidDragNDropImage(url))
+                    cropImage(url)
+                else
+                    errorDialog.open()
```

Comment:
Yeah, still cannot grasp how this is valid code, esp. after switching from C++ 😁 I'd rather not though; this will work until someone adds more code below, after which it's easy to miss the early return

Reply:
It's pretty common to early return in JS, but as you wish